static/js/narrow.js

```diff
@@ -228,13 +231,20 @@ exports.activate = function (raw_operators, opts) {
         muting_enabled,
     });
 
+    const key = hash_util.operators_to_hash(operators);
```

Comment:
Given that `MessageListDataCache` already knows how to use `hash_util`, it seems like we could tweak its interface here so that `narrow.js` doesn't have to understand this particular detail?

More broadly, this section of code just feels kinda clumsy, as we are first trying to create a `MessageListData` as `msg_data`, but then **after** that, we try to get an MLD from the cache, and we overwrite `msg_data` then.